File: openmp/src/main/java/com/example/openmp/NativeLib.java
package com.example.openmp;

public class NativeLib {

    // Used to load the 'openmp' library on application startup.
    static {
        System.loadLibrary("openmp");
    }

    /**
     * A native method that is implemented by the 'openmp' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();
}

File: nativelib/src/main/java/com/example/nativelib/NativeLib.kt
package com.example.nativelib

import java.io.ByteArrayInputStream
import java.io.File

class NativeLib {

    /**
     * A native method that is implemented by the 'nativelib' native library,
     * which is packaged with this application.
     */
    external fun stringFromJNI(): String

    external fun test(file: File, inputStream: ByteArrayInputStream)

    companion object {
        // Used to load the 'nativelib' library on application startup.
        init {
            System.loadLibrary("nativelib")
        }
    }
}